### ISSUE TITLE
Experiment Module: Limiting E-mail Alerts

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#395]
Following issue #299 we have now implemented e-mail alerts for the experiment module, but these alerts are created every time a change is made.

Speaking with the FFEC sub-group last week, this is apparently how fieldnet worked and they are happy to keep the system as it is for the time being, accepting that there will be a lot of these alerts in the months where things are being updated/ requested.

However, I want to create a holding issue for this in case they change their mind once the system is in place.

Previously @paul121 and I discussed having a minimum alert structure (for example when a status change occurs), but including the option to send an alert above and beyond this if necessary.

Duplicate issue for #506

**Comment by aislinnpearson** [comment]
Hi @paul121, @mstenta 

So I have been thinking about this in terms of user experience/ functionality and I have a solution I think would work from that perspective, but I'd need a lot of insight from you in terms of implementation.

What I'd lie to do is keep the solution as simple but as flexible as possible. 

1. The alert options would have to be associated with user accounts instead of researchers for two reasons: (1) Some of our users don't have researcher profiles (operators and farm managers usually), yet the farm managers especially will need alerts and (2) I don't think people who don't have a FarmOS account should be getting emails. This is the way it currently works because e-mails are associated with researchers instead of users, a this means I've had people setting up accounts for other people who don't get told, have no idea what the e-mails are for and most importantly can't log in to switch them off. There is an issue for that here: #589 
2. The ideal situation would be that people can select which assets/ entities (proposals/ experiments only as this is where you are named) they get alerts for, with the option to include alerts for logs associated with those experiments or assets.
3. The users can choose to get alerts on the chosen assets daily, weekly or monthly.
4. I need to think about the content of the messages (I have a separate issue for this here #506) but I see them being either:  (a) a confirmation email to say person x (could be yourself) has added your name to a new asset/ entity, please click here to receive alerts about that asset/ entity; (b) a list of any new logs that have been added since your last alert for each entity you choose to follow; (c) any data fields have been updated for the logs/ entities you follow and this is what it now says. (a) would be a standalone email but (b) and (c) could be combined into one.
5. The default functionality is you get email (a) if your name is on an asset or entity but you only get e-mails (b) and (c) if you choose to.
6. There are three separate tick boxes for all logs (farm mangers) and all experiments (data admins/ trials managers)

As an aside, one of the data stewards pointed out to me that the site admins get _every_ email for the experiment entitles. Ideally we'd switch them off at the same time so they get the same options as everyone else.

Sorry, I realise this creates a nice simple user interface but might be pretty complicated to implement. I suspect you can see now why we need to discuss the reality of implementation.

**Comment by aislinnpearson** [comment]
Speaking with @paul121 about this on 24/11/2023 we talked through the complexity and agreed the following to simplify the initial release:

- We _won't_ include assets (this can come later if there is a user need)
- We _won't_ have an opt in function for email (a) but rather a opt out function as this is easier to build
- We will look into creating a page that lists what you are following, what you are associated with (but not following) and then an option to add any other experiment/ proposal
- If we have to pick between experiments and proposals, pick Proposals as they are linked to experiments (this is somewhat unresolved)
- We could have an 'auto-opt in' function so that you are automatically get alerts if added to a trial
- The two buttons for 'follow all logs' and 'follow all experiments' is fine
- We'll start with the most simple option for getting edits _and_ logs, and then see how it goes. If people complain a lot we will refine the types of alerts you get.
- You will get alerts about your own logs.

Daily summary: would be anything sent at the end of the day

**Comment by aislinnpearson** [comment]
points from 28/11/23

First step: move alerts from Researchers to users, and Aislinn to create issue for this. We can then com back and add additional issues as we go through.

Setting daily/ weekly/ monthly for different entities is complex.
Currently the system sends an email as something happens. To queue means save information to do something later. This would be a query against time stamps.
Alternatively could have a page that shows you what has changed. unique for each user. Page could update and be visible at any time. But equally could become a very long list that gets ignored.

**Comment by paul121** [comment]
> First step: move alerts from Researchers to users, and Aislinn to create issue for this. We can then com back and add additional issues as we go through.

Done!

I've also made a simple change to stop sending admins *all* notification emails for now. This will be included in the 2.18 release.

**Comment by paul121** [comment]
Closing this issue - see email tracking issue for more pending improvements to emails: https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/506
